### PR TITLE
Add structs to support proxy server configuration

### PR DESCRIPTION
**Pull request by Bhashinee**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

Ubuntu 16.04
Proxy server - squid

### CODE REVIEW

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java [hunk 1]
```diff
@@ -325,6 +328,33 @@ private void populateSenderConfigurationOptions(SenderConfiguration senderConfig
                 senderConfiguration.setParameters(clientParams);
             }
         }
+
```
Comment (chamil321) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java]: Shall we remove this empty line?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java [hunk 2]
```diff
@@ -325,6 +328,33 @@ private void populateSenderConfigurationOptions(SenderConfiguration senderConfig
...
+            String proxyUserName = proxy.getStringField(Constants.PROXY_USER_NAME_INDEX);
+            String proxyPassword = proxy.getStringField(Constants.PROXY_PASSWORD_INDEX);
+
+            if (proxyHost != null && proxyPort != 0) {
+                try {
+                    proxyServerConfiguration = new ProxyServerConfiguration(proxyHost, proxyPort);
+                } catch (UnknownHostException e) {
+                    throw new BallerinaConnectorException("Failed to resolve host" + proxyHost, e);
+                }
+
```
Comment (chamil321) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java]: Shall we remove this empty line?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java (hunk 1 above)
Comment (Bhashinee) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java]: Done

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java (hunk 2 above)
Comment (Bhashinee) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java]: Done

**Pull request merged**